portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java

```
@@ -1451,7 +1486,49 @@ public Layout getLayoutByIconImageId(long iconImageId)
 	public List<Layout> getLayouts(
 		long groupId, boolean privateLayout, String type) {
 
-		return layoutPersistence.findByG_P_T(groupId, privateLayout, type);
+		Group group = groupPersistence.fetchByPrimaryKey(groupId);
+
+		List<Layout> layouts = layoutPersistence.findByG_P_T(
+			groupId, privateLayout, type);
+
+		if (!group.isUser()) {
+			return layouts;
+		}
+
+		layouts = new ArrayList<>(layouts);
+
+		Set<Long> checkedPlids = new HashSet<>();
+		Queue<Long> checkParentLayoutIds = new ArrayDeque<>();
+
+		checkParentLayoutIds.add(LayoutConstants.DEFAULT_PARENT_LAYOUT_ID);
+
+		while (!checkParentLayoutIds.isEmpty()) {
+			long parentLayoutId = checkParentLayoutIds.poll();
+
+			LayoutSet layoutSet = layoutSetPersistence.fetchByG_P(
+				groupId, privateLayout);
+
+			try {
+				List<Layout> userGroupLayouts = _addUserGroupLayouts(
+					group, layoutSet, Collections.emptyList(), parentLayoutId);
+
+				for (Layout userGroupLayout : userGroupLayouts) {
+					long userGroupPlid = userGroupLayout.getPlid();
+					long userGroupLayoutId = userGroupLayout.getLayoutId();
+
+					if (checkedPlids.add(userGroupPlid)) {
+						layouts.add(userGroupLayout);
+
+						checkParentLayoutIds.add(userGroupLayoutId);
+					}
+				}
+			}
+			catch (PortalException pe) {
```

Maybe we should just throw the exception.